spring-cloud-context/src/main/java/org/springframework/cloud/context/named/NamedContextFactory.java

```diff
@@ -126,33 +161,20 @@ protected AnnotationConfigApplicationContext createContext(String name) {
 			else {
 				beanFactory.setBeanClassLoader(parent.getClassLoader());
 			}
-			context = new AnnotationConfigApplicationContext(beanFactory);
-			context.setClassLoader(this.parent.getClassLoader());
+			context = AotDetector.useGeneratedArtifacts() ? new GenericApplicationContext(beanFactory)
+					: new AnnotationConfigApplicationContext(beanFactory);
+			context.setClassLoader(parent.getClassLoader());
 		}
 		else {
-			context = new AnnotationConfigApplicationContext();
-		}
-		if (this.configurations.containsKey(name)) {
-			for (Class<?> configuration : this.configurations.get(name).getConfiguration()) {
-				context.register(configuration);
-			}
+			context = new GenericApplicationContext();
```

Comment:
Should this get the AotDetector treatment as well?

Reply:
Good catch. I think for child contexts we always have the parent, but maybe there's some edge-case where we don't... I guess better to have it there as well.